Update to latest NPAPI Flash

rshuler opened the pull request:
Adobe is now supporting the latest version for NPAPI.
Previously, for the past few years they had been only adding
security updates for the NPAPI build.

As part of this, the download path and file structure has changed,
eliminating the need to unzip an inner tar file (and significnatly
reducing the download file size, since we can now download just
the 64-bit binary we need).

https://phabricator.endlessm.com/T15713

Commit:
Update to latest NPAPI Flash

Adobe is now supporting the latest version for NPAPI.
Previously, for the past few years they had been only adding
security updates for the NPAPI build.

As part of this, the download path and file structure has changed,
eliminating the need to unzip an inner tar file (and significnatly
reducing the download file size, since we can now download just
the 64-bit binary we need).

https://phabricator.endlessm.com/T15713

mariospr commented on firefox-plugins-installer:
```diff
@@ -43,11 +43,11 @@ JAVA_LINK=${PLUGINS_DIR}/libnpjp2.so
-FLASH_VERSION=11.2.202.644
-FLASH_SHA256SUM=aba0e2f5940fc2588357e93f7bad84a0db51792b2f2bd1d24be9ef491e87c041
-FLASH_FILENAME=fp_${FLASH_VERSION}_archive.zip
+FLASH_VERSION=24.0.0.221
```
This will break as soon as Adobe decides to ship newer version of the plugin, as they tend to remove the previous version from the servers.

Maybe we could leverage the LKGV file in https://s3-us-west-2.amazonaws.com/abacadaba/flashplugin-x86_64.lkgv for this too, and add the necessary code here to check it and build the actual URL based on it? (replacing `ppapi` with 
`npapi` to get the right URL for firefox).

Or even, if we want to make it more independent from the PPAPI plugin, we should extend the flashplugin-x86_64.lkgv file so that we include there the latest versions of the PPAPI **and** the NPAPI plugin, along with the extra data (URL, SHA256SUM), and also update the [jenkins-check-adobe-flash-plugin script](https://github.com/endlessm/eos-build/blob/master/ci-integration/jenkins-check-adobe-flash-plugin) to check for changes in both versions too. That way it should be more robust and easier to handle in the future, without requiring OS upgrades.

rshuler replied:
Good point.  Since we would need the sha256sum, I think the second approach is the way to go.  We just need to be careful to make the change in such a way that we don't break existing users of the chrome plugin updater.  I assume as long as we keep the existing "URL" and "SHA256SUM" variable names for PPAPI and just add new strings for NPAPI, we should be fine, correct?

mariospr replied:
Correct, although it's kind of ugly to have that distinction, it would be nicer if we could have something more consistent, like using a prefix for the type, but I suppose something like this would not be too bad:
```
VERSION=<version>
URL=<url>
SHA256SUM=<sha256sum>

NPAPI_VERSION=<version>
NPAPI_URL=<url>
NPAPI_SHA256SUM=<sha256sum>
```
...even though I'd like it more if the first keys were prefixed by `PPAPI_` but, as you say, that would break current users so that's probably not a good idea, unless we also keep the former keys around for backwards compatibility for a while, as an upgrade path.

But that's probably too much of a hassle for something as simple as adding another possibility (and likely the only additional one we will ever need), so I think the unprefixed version for PPAPI plugins should be fine.

mariospr commented on firefox-plugins-installer:
```diff
@@ -43,11 +43,11 @@ JAVA_LINK=${PLUGINS_DIR}/libnpjp2.so
-FLASH_FILENAME=fp_${FLASH_VERSION}_archive.zip
+FLASH_VERSION=24.0.0.221
+FLASH_SHA256SUM=087048af5fa3c46a8c283403a341d4684ef4b966f4e407195b43c24bf5ae64b1
+FLASH_FILENAME=flash_player_npapi_linux.x86_64.tar.gz
```
Not a big deal, but we use `$(arch)` for the Chrome plugin downloader, and it would probably be nice to use it here too.

rshuler replied:
Makes sense.

Commit:
Query the LKGV file in S3 for the flash plugin and enable updates

At the moment, Firefox was installing one version of Flash and sticking
to it forever, which was a reasonable approach considering that Adobe
was not releasing new versions of the NPAAPI plugin for Linux. However,
that has recently changed and Adobe is now releasing versions of this
plugin more regularly, so it makes sense not only to make sure it gets
installed but also that it gets regularly updated.

To implement this, we borrowed some code of the updater we already use
for chromium and do a similar check for new versions (every 7 days, max)
and update it locally if needed. Also, to prevent depending on OS upgrades
just to update the latest version available from Adobe (which tends to
remove the old one from their servers after a few days), we also poke
out LKGV file in S3 (which we can update separately) first, and only
if that file or the relevant fields are not found, we use the fallback
values hardcoded in the script.

Last, note that the code added here never calls exit_with_error(), which
is done on purpose so that a failure installing the flash plugin does
not prevent the script from installing the Java plugin, just afterwards.

https://phabricator.endlessm.com/T15713
